modules/webhook/slack.go

```diff
@@ -233,6 +233,8 @@ func getSlackPushPayload(p *api.PushPayload, slack *SlackMeta) (*SlackPayload, e
 		IconURL:  slack.IconURL,
 		Attachments: []SlackAttachment{{
 			Color: slack.Color,
+			Title: p.Repo.HTMLURL,
```

Comment:
An URL for a title? I expected some plain text (https://api.slack.com/reference/messaging/attachments).

Reply:
Do you have a suggestion for that text? Other payloads have issue titles or some such, but I didn't know of a good one for this, unless I just re-use `x new commits pushed` or something.

Reply:
The repository payload seems to do something similar.

```go
func getSlackRepositoryPayload(p *api.RepositoryPayload, slack *SlackMeta) (*SlackPayload, error) {
	senderLink := SlackLinkFormatter(setting.AppURL+p.Sender.UserName, p.Sender.UserName)
	repoLink := SlackLinkFormatter(p.Repository.HTMLURL, p.Repository.FullName)
	var text, title, attachmentText string

	switch p.Action {
	case api.HookRepoCreated:
		text = fmt.Sprintf("[%s] Repository created by %s", repoLink, senderLink)
		title = p.Repository.HTMLURL
	case api.HookRepoDeleted:
		text = fmt.Sprintf("[%s] Repository deleted by %s", repoLink, senderLink)
	}

	return &SlackPayload{
		Channel:  slack.Channel,
		Text:     text,
		Username: slack.Username,
		IconURL:  slack.IconURL,
		Attachments: []SlackAttachment{{
			Color:     slack.Color,
			Title:     title,
			TitleLink: title,
			Text:      attachmentText,
		}},
	}, nil
}
```

Reply:
No idea, but if the attachment links to the repo itself (which is weird, maybe?), perhaps the repo name should be used?

I'm under the impression that it should better to remove the property from the payload when empty rather than sending improper data. 🤔

Reply:
I'd agree with not sending anything, but that seems to be what caused the bug for whatever reason.

It could be repo name, but even that is sort of dupe data. Better than a URL, though? 😅

Reply:
I'm not a webhook user; not even a Slack user, so please take my opinion with a grain of salt. From what I [read](https://api.slack.com/reference/messaging/attachments) from the [docs](https://api.slack.com/messaging/composing/layouts#when-to-use-attachments), I think we're using the attachment section to produce a "Read more" action. It looks like it should not have a link at all, and if any title, it should be.... "More details" or something like that (but it would be redundant because I bet Slack already gives you the "Read more").

Reply:
Bars on my screenshots above, "see more" works in RC but possibly not Slack, since RC allows you to hide it but Slack appears to show all of it no matter what

Reply:
But I am also not a Slack or RC user, I just have some workspaces for testing webhooks lately.

Reply:
https://api.slack.com/messaging/composing/layouts#when-to-use-attachments

>
> Any content displayed within attachments may be wrapped, truncated, or hidden behind a "show more" style option by Slack clients. 

So I guess it depends on the client.

Reply:
For the record I am not against changing this, I just really can't think of a good title.  
An empty title would break the webhook (at least for RC, not sure about Slack or mattermost)

Reply:
Why not open it up as an option with a default? Maybe more trouble than it's worth?

Reply:
Since it only applies to this payload, I don't know that it would be worth it.  
Other payloads have something meaningful for this title.

Reply:
What about "details"? The problem is of course l10n....
Anyway, I'm not in the least against leaving it like this if nobody complains. 😄

Reply:
Agreed. Maybe it can be updated in future PR.